### ISSUE TITLE
default config, don't try to unwrap if the toml not found

### ISSUE THREAD

**Issue by thedavidmeister** [riker-rs/riker#13]
I'm getting this if i haven't setup the config toml:

```
thread 'tests::can_instantiate' panicked at 'called `Result::unwrap()` on an `Err` value: configuration file "config/riker.toml" not found', libcore/result.rs:945:5
```

it seems to me that default fallback config could work better than an unwrap

**Comment by leenozara** [comment]
@thedavidmeister Actually I stand corrected. In Akka it is possible to start an application without a config file. Only the individual parameters should fail if they don't exist.

Let me review this a little more. Just FYI I tend to use Akka as a reference since I'm familiar with it and it is well designed. I guess in this instance I'm not so familiar :).

**Comment by thedavidmeister** [comment]
yes it certainly seems like one of those things that a beginner like me would run into early, but someone more experienced wouldn't even notice

i also found it awkward that i had to symlink the config into place to support a repo with several crates in it, even for crates that did not use riker but depended on crates that did (so there was a copy of the same config in both crates).

**Comment by thedavidmeister** [comment]
i also haven't tested this, but the error i saw was along the lines of "file not found" not "config not found"

this would mean that i can provide all the config directly in the code but then need to ship an empty config file, which doesn't seem right even given the riker philosophy

**Comment by leenozara** [comment]
@thedavidmeister There's a PR `config-fix` that addresses your issue:

`riker.toml` is optional. If it is not provided then default values for the bare minimum settings are used.

Again, thanks so much for your help pointing this out and working with me to design a nice fix for beginners. If there's anything else you see that needs improving, especially from a beginner's perspective please let me know and I'll jump on it.

Let me know if this fixes your issue and we can close it. Thanks!

**Comment by leenozara** [comment]
@thedavidmeister 

Regarding your symlinking. It's possible to override the location using an env variable:

- `RIKER_CONF` to override the full path to `riker.toml`
- `APP_CONF` to override the full path to `app.toml`

Example:

`export RIKER_CONF=/var/conf/riker/riker.toml`.

This feature is already there but not documented. I'll add an issue to add this to the documentation.

**Comment by leenozara** [comment]
Closing this as it was merged into `0.1.5`
